lib/http/options.rb

```diff
@@ -6,6 +6,19 @@
 require "http/uri"
 
 module HTTP
+  # Options are passed to request as a hash.
+  #
+  # Keys for the hash include:
+  #
+  #   :headers for request headers
+  #
+  #   :body to set the body contents
+  #
+  #   :json to send JSON
+  #
+  #   :form to send form parameters, also used with multipart form using HTTP::FormData::File
+  #
+  #   :ssl for TLS/SSL options, @see https://ruby-doc.org/stdlib-2.5.1/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html OpenSSL::SSL::SSLContext
```

Comment:
Note that I had to step through the code over many attempts and many hours to discover that the key for the options to set the TLS/SSL verify mode or path to the CA file was `:ssl`.

Reply:
:ssl does not related to SSLContext, for that you need `:ssl_context`. I tend to think that this should be done via YARD dsl macro.

Reply:
It took hours of stepping through the code to know what the `:ca_file` key needed to be.  Only after finding that `:ssl` is passed to that class and seeing its attributes did I discover that it needed to be `:ca_file`.

Reply:
I'm not saying it shouldn't be documented.  I'm saying that we must document both. And don't confuse ssl option with sslcontext